src/ProjectNeon/Assets/Features/Effects/EffectOnAttacker.cs

```diff
@@ -0,0 +1,31 @@
+﻿using UnityEngine;
+using System.Collections;
+
+/**
+ * Effects that applies to the attacker on an attack event
+ */
+public class EffectOnAttacker : Effect
+{
+    private Effect _effect;
+    private Member _performer;
+    private Target _effectTarget;
+
+    public EffectOnAttacker(Effect damage)
+    {
+        _effect = damage;
+        BattleEvent.Subscribe<Attack>(attack => Execute(attack), this);
+    }
+
+    void Effect.Apply(Member source, Target target)
+    {
+        _performer = source;
+        _effectTarget = target;
+    }
+
+    void Execute(Attack attack)
+    {
+        if (attack.Attacker.Equals(_performer))
```

Comment:
This condition looks wrong. 

Why would the target also expect to be the attacker? That makes no sense.

Reply:
@SilasReinagel So we don't need to check anything here, the attacker will always be defined

Reply:
@paulodamaso I don't understand.

One must filter events, since not all Attacks are attacking someone who has Aura of Retribution. 

If the Solider current has Aura of Retribution, and enemy Red Drone attacks the Paladin, Soldier shouldn't deal Retribution damage to the Red Drone.

Reply:
@SilasReinagel UNderstood the problem, will fix

Reply:
@SilasReinagel This condition is already being tested in `DamageOnAttacker_ApplyEffect_AttackerIsNotDamagedOnAttackingOther()` please take a look